src/ui.rs

```diff
@@ -158,12 +167,16 @@ impl Ui {
 
     /// UI for a single target. Must be run on its own thread.
     ///
-    /// Returns if the target was successful or not>
-    fn target_ui(updates: Receiver<Output>, target: String, show_cmd: bool) -> bool {
+    /// Returns a TargetUiRc
+    fn target_ui(updates: Receiver<Output>, target: String, show_cmd: bool) -> TargetUiRc {
```

Comment:
Could we return `Result<Option<i32>>` instead and avoid kinda the same thing with `TargetUiRc`?

Reply:
I also thought of returning a `Result` instead. What confuses me here is that we keep on incrementing the error counter instead of returning directly. Does that mean that we actually need to let the whole state machine go through? Or could we return as soon as an error is detected (and forwarded it up the stack at the same time)?
On the other hand, it seems the error is more conveniently consume as part of the `Stage` terminal.

If we need to go through the whole loop, then probably the error will be quite generic.

Reply:
Ah I see what you mean. This makes sense, then